pandas/tests/io/pytables/test_store.py

```diff
@@ -288,35 +288,47 @@ def test_store_dropna(tmp_path, setup_path):
     # # Test to make sure defaults are to not drop.
     # # Corresponding to Issue 9382
     path = tmp_path / setup_path
-    df_with_missing.to_hdf(path, "df", format="table")
+    df_with_missing.to_hdf(path, key="df", format="table")
     reloaded = read_hdf(path, "df")
     tm.assert_frame_equal(df_with_missing, reloaded)
 
     path = tmp_path / setup_path
-    df_with_missing.to_hdf(path, "df", format="table", dropna=False)
+    df_with_missing.to_hdf(path, key="df", format="table", dropna=False)
     reloaded = read_hdf(path, "df")
     tm.assert_frame_equal(df_with_missing, reloaded)
 
     path = tmp_path / setup_path
-    df_with_missing.to_hdf(path, "df", format="table", dropna=True)
+    df_with_missing.to_hdf(path, key="df", format="table", dropna=True)
     reloaded = read_hdf(path, "df")
     tm.assert_frame_equal(df_without_missing, reloaded)
 
 
+def test_keyword_deprecation():
+    # GH 54229
+    msg = (
+        "Starting with pandas version 3.0 all arguments of to_hdf except for the "
+        "argument 'path_or_buf' will be keyword-only."
+    )
+    df = DataFrame([{"A": 1, "B": 2, "C": 3}, {"A": 1, "B": 2, "C": 3}])
+
+    with tm.assert_produces_warning(FutureWarning, match=msg):
+        df.to_hdf("example", "key")
```

Comment:
this test is leaving behind an "example" file.  can you do a follow-up to use tm.ensure_clean to avoid this

Reply:
@jbrockmendel sure